src/windows7.rs

```diff
@@ -0,0 +1,23 @@
+use crate::Error;
+use core::{ffi::c_void, mem::MaybeUninit};
+
+type BOOLEAN = u8;
+type ULONG = u32;
+const TRUE: BOOLEAN = 1;
+
+#[link(name = "advapi32")]
+extern "system" {
+    #[link_name = "SystemFunction036"]
+    fn RtlGenRandom(RandomBuffer: *mut c_void, RandomBufferLength: ULONG) -> BOOLEAN;
+}
+
+pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
+    // Prevent overflow of ULONG
```

Comment:
Even when building for a win7 target, this is actually probably going to be running on Windows 10.

Especially with that in mind, but regardless, we should document here why we don't try `BCryptGenRandom` before `RtlGenRandom` like we did before. It seems to be like in the abstract we should keep the same logic we had before. One concrete reason to avoid trying `BCryptGenRandom` before `RtlGenRandom` is that we would end up never testing the `RtlGenRandom` code path because we have no runtime environment that would trigger the fallback logic.

Regardless of what we do, we need to update the documentation about what we use on each of win7 and regular *-windows-* targets as part of this change.

Reply:
Added a bunch of comments explaining the different options and why we do what we do.